router.go

```diff
@@ -192,3 +204,14 @@ func (m MapRouter) Choose(ctx context.Context, req *transport.Request) (transpor
 
 	return transport.HandlerSpec{}, yarpcerrors.Newf(yarpcerrors.CodeUnimplemented, "unrecognized procedure %q for service %q", req.Procedure, req.Service)
 }
+
+// Extract keys from service names map and return a formatted string
+func getAvailableServiceNames(svcMap map[string]struct{}) string {
+	var strServiceNames []string
```

Comment:
nit: Just call this `serviceNames`. We don't need to include the type name in the variable name.